oneanddone/tasks/views.py

```diff
@@ -350,19 +351,125 @@ def post(self, *args, **kwargs):
         return redirect('tasks.feedback', attempt.pk)
 
 
-class TaskDetailView(generic.DetailView):
-    model = Task
-    template_name = 'tasks/detail.html'
-    allow_expired_tasks = True
+class TaskAttemptDetailView(LoginRequiredMixin,
+                            MyStaffUserRequiredMixin,
+                            generic.View):
+
+    def get(self, request, *args, **kwargs):
+        view = TaskAttemptDisplayView.as_view()
+        return view(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        view = SubmitAdminVerificationResponseView.as_view()
+        return view(request, *args, **kwargs)
+
+
+class TaskAttemptDisplayView(LoginRequiredMixin,
+                             MyStaffUserRequiredMixin,
+                             generic.DetailView):
+    model = TaskAttempt
+    template_name = 'tasks/attempt_detail.html'
 
     def get_context_data(self, *args, **kwargs):
-        ctx = super(TaskDetailView, self).get_context_data(*args, **kwargs)
+        ctx = super(TaskAttemptDisplayView, self).get_context_data(*args, **kwargs)
+        ctx['form'] = SubmitVerifiedTaskForm()
+        return ctx
+
+
+class SubmitAdminVerificationResponseView(LoginRequiredMixin,
+                                          MyStaffUserRequiredMixin,
+                                          generic.detail.SingleObjectMixin,
+                                          generic.FormView):
+
+    form_class = SubmitVerifiedTaskForm
+    model = TaskAttempt
+    template_name = 'tasks/attempt_detail.html'
+
+    def post(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        return super(SubmitAdminVerificationResponseView, self).post(request, *args, **kwargs)
+
+    def form_valid(self, form):
+
+        attempt = self.get_object()
+
+        communication = form.save(commit=False)
+        communication.attempt = attempt
+        communication.type = TaskAttemptCommunication.ADMIN
+        communication.creator = self.request.user
+        communication.save()
+
+        # Send email to attempt user
+        if attempt.user.profile.consent_to_email:
+            task_name = attempt.task.name
+            subject = 'One and Done task verification update for %s' % task_name
```

Comment:
Good point. As you've no doubt noticed I often forget about i18n. Glad to hear that you think this is acceptable. :)